platform/linuxbsd/freedesktop_screensaver.cpp

```diff
@@ -0,0 +1,123 @@
+/*  freedesktop_screensaver.cpp                                          */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "freedesktop_screensaver.h"
+
+#ifdef DBUS_ENABLED
+
+#include "core/config/project_settings.h"
+#include "dbus/dbus.h"
+
+#define BUS_OBJECT_NAME "org.freedesktop.ScreenSaver"
+#define BUS_OBJECT_PATH "/org/freedesktop/ScreenSaver"
+#define BUS_INTERFACE "org.freedesktop.ScreenSaver"
+
+void FreeDesktopScreenSaver::inhibit() {
+	if (unsupported) {
+		return;
+	}
+
+	DBusError error;
+	dbus_error_init(&error);
+
+	DBusConnection *bus = dbus_bus_get(DBUS_BUS_SESSION, &error);
+	if (dbus_error_is_set(&error)) {
+		unsupported = true;
+		return;
+	}
+
+	String app_name_string = ProjectSettings::get_singleton()->get("application/config/name");
+	const char *app_name = app_name_string.is_empty() ? "Godot Engine" : app_name_string.utf8().get_data();
+
+	const char *reason = "Playing game";
```

Comment:
Maybe should be something like
```suggestion
	const char *reason = "Playing Godot Engine project";
```
to also encompass non-games?

Reply:
Not sure about this one... I get the non-games point, but consider playing some game you downloaded from Steam (I _assume_ most Godot projects are games) and seeing "Playing Godot Engine project" in an operating system menu (honestly not exactly sure where different desktop environment expose this string). I feel like that goes against the usual white-label approach of game engines/UI toolkits and it might also confuse some less technical users.

Reply:
Can you use `app_name` to make it specific? E.g.:
```
const char *reason = "Playing " + app_name;
```

Reply:
I'm already using `app_name` for the `application_name`, so that would show the same string twice. I think I'm just going with `Running Godot Engine project`. Considering the few places this string is displayed in and the different ways developers are using this `reason` parameter already (I looked through multiple pages of [this](https://grep.app/search?q=org.freedesktop.ScreenSaver)) it just doesn't seem to matter that much.